### PR TITLE
fix(dpi): read site DPI counters from stat/sitedpi

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** `get_dpi_statistics` and `list_top_applications` always
report zero applications while looking healthy: the `stat/dpi` path they
query answers with nothing on current controllers.

**Root cause + fix:** the counters live in the `stat/sitedpi` report
(POST with type `by_app` / `by_cat`) — route verified live on Network
10.5.67. Rows are returned as the controller reports them (`app`/`cat`
numeric catalog ids, byte/packet counters) with a computed `total_bytes`,
sorted descending; ids are translatable via `list_dpi_applications` /
`list_dpi_categories`.

Two honesty changes:

- The `time_range` parameter is removed from both tools: the counters are
  lifetime totals and no variant of the endpoint accepts a window — the
  parameter validated six values and changed nothing.
- Current releases run traffic identification through the flow engine
  instead of the classic counters, so `stat/sitedpi` legitimately answers
  with the empty shape `data:[{}]` there. That state now comes back with a
  `note` pointing at the flow tools (`get_top_flows`,
  `get_flow_analytics`) rather than a silent empty.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

`time_range` is removed from two tool signatures; it was accepted and
ignored before, so no caller's results change.

## Testing

- Tests rebuilt around the real `stat/sitedpi` request/response shapes
  (`by_app`/`by_cat` POST bodies, numeric ids, `total_bytes` computation
  and ordering, the empty-with-note state).
- `pytest tests/unit/` and `pre-commit run --all-files` pass.
- `API.md` updated: parameter removed, response examples now show the
  controller's numeric catalog ids and the flow-engine note.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API.md, CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; route and empty-shape behavior
verified against a live Network 10.5.67 controller.
**Security Review:** Completed — all fixture values are synthetic.

### CODE REVIEW

File: src/tools/dpi.py [hunk 1]
```diff
@@ -16,126 +16,103 @@
 async def get_dpi_statistics(
     site_id: str,
     settings: Settings,
-    time_range: str = "24h",
 ) -> dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/dpi.py:19]: Fixed in 93f1d27 — the integration suite now calls the tools as they are, asserts the numeric `app` catalog ids, and the two time-range tests are replaced with a check that a flow-engine controller's empty counters carry the note pointing at the flow tools. On keeping `time_range` as a deprecated no-op: that parameter is what this PR removes on purpose — it validated six values and changed nothing — and `application`/`category` aliases would invent names the endpoint doesn't report.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
